dataland-frontend/tests/e2e/fixtures/p2p/P2pDataFixtures.ts

```diff
@@ -41,289 +39,290 @@ export function generateP2pFixtures(
- * @param undefinedProbability the ratio of fields to be undefined (number between 0 and 1)
- * @returns a random P2pRealEstate
- */
-export function getSectorRealEstate(undefinedProbability: number): P2pRealEstate {
-  return {
-    buildingEfficiency: {
-      buildingSpecificReburbishmentRoadmap: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-      zeroEmissionBuildingShare: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-      buildingEnergyEfficiency: valueOrUndefined(randomInt(10000), undefinedProbability),
-    },
-    energySource: {
-      renewableHeating: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-    },
-    technology: {
-      useOfDistrictHeatingNetworks: valueOrUndefined(randomYesNo(), undefinedProbability),
-      heatPumpUsage: valueOrUndefined(randomYesNo(), undefinedProbability),
-    },
-  };
-}
+  sectorPresent(sector: P2pSector): boolean {
+    return this.sectors.indexOf(sector) != -1;
+  }
 
-/**
- * Generates random P2pSteel instance
- * @param undefinedProbability the ratio of fields to be undefined (number between 0 and 1)
- * @returns a random P2pSteel
- */
-export function getSectorSteel(undefinedProbability: number): P2pSteel {
-  return {
-    energy: {
-      emissionIntensityOfElectricity: valueOrUndefined(randomInt(10000), undefinedProbability),
-      greenHydrogenUsage: valueOrUndefined(randomYesNo(), undefinedProbability),
-    },
-    technology: {
-      blastFurnacePhaseOut: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-      lowCarbonSteelScaleUp: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-    },
-  };
-}
+  /**
+   * Method to generate the sectors for a p2p dataset
+   * @param toggleRandomSectors determines if the sector list should include all possible sectors or a randomized selection
+   * @returns the sectors used for generating fixtures
+   */
+  generateSectors(toggleRandomSectors: boolean): P2pSector[] {
+    if (toggleRandomSectors) {
+      return faker.helpers.arrayElements(Object.values(P2pSector));
+    } else {
+      return Object.values(P2pSector);
+    }
+  }
 
-/**
- * Generates random P2pFreightTransportByRoad instance
- * @param undefinedProbability the ratio of fields to be undefined (number between 0 and 1)
- * @returns a random P2pFreightTransportByRoad
- */
-export function getSectorFreightTransportByRoad(undefinedProbability: number): P2pFreightTransportByRoad {
-  return {
-    technology: {
-      driveMix: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-      icePhaseOut: valueOrUndefined(randomFutureDate(), undefinedProbability),
-    },
-    energy: {
-      fuelMix: valueOrUndefined(randomPercentageValue(), undefinedProbability),
-    },
-  };
-}
+  /**
+   * Generates random general information for the P2P Framework
+   * @returns a random P2pGeneral instance with the given sectors
+   */
+  getSectorGeneral(): P2pGeneral {
```

Comment:
This always returns an object. So maybe "generateSectorGeneral()" per our naming convention?

Reply:
fixed